### ISSUE TITLE
ClassDeclaration cannot contain the given type directly

### ISSUE THREAD

**Issue by niieani** [codemix/flow-runtime#87]
This is a:

 - [x] Bug Report

Which concerns:

 - [x] flow-runtime
 - [x] flow-runtime-validators

---

### What is the current behaviour?

This currently throws "ClassDeclaration cannot contain the given type directly", even though the code is 100% correct. Also, the file does not contain `// @flow` pragma, but it still gets checked nonetheless.

```js
const someMethodName = 'someName'

function mixinFactory(Component: Function) {
  return class ComponentInTree extends Component {
    [someMethodName]() {
      
    }
  }
}

mixinFactory(class {})
```

---

### What is the expected behaviour?

No errors should be thrown, especially when not using assertions.  It would also be nice if files without the `// @flow` pragma would not transpile with flow-runtime enable by default.

---

### Which package versions are you using?
0.10.0 (latest)

**Comment by phpnode** [comment]
fixed in 0.11.1

**Comment by niieani** [comment]
Thank you!
